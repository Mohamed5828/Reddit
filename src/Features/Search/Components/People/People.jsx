import PeopleCardItem from "../PeopleCardItem/PeopleCardItem";
import { PeopleContainer } from "./People.styled";

/**
 * Component that contains the People Card Elements.
 *
 * @Component
 * @param {object} PeopleList -  contains the People
 * @param {object} PeopleFollow -  contains the People you are following
 * @returns {React.Component}
 */
const People = ({ PeopleList, PeopleFollow }) => {
  if (PeopleList && PeopleFollow) {
    return (
      <PeopleContainer>
        <p className="title">People</p>
        {PeopleList.map((People) => (
          <PeopleCardItem
            avatar={People.avatar}
            key={People.userID}
            userID={People.userID}
            about={People.about}
            totalKarmas={People.totalKarmas}
            username={People.username}
            isFollow={PeopleFollow.find((element) => {
              return element.id === People.userID;
            })}
          />
        ))}
      </PeopleContainer>
    );
  }
};

export default People;
