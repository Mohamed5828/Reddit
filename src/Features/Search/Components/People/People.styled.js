import styled from "styled-components";
export const PeopleContainer = styled.div`
  background-color: ${({ theme }) => theme.background.primary};
  border: thin solid ${({ theme }) => theme.lineColor.primary};

  border-radius: 4px;
  box-shadow: 0 2px 4px 0 rgb(0 0 0 / 6%);
  margin-bottom: 16px;
  max-width: 100%;
  & .title {
    font-size: 16px;
    line-height: 20px;
    color: ${({ theme }) => theme.color.titleText_of_link};
    padding: 16px 16px 0;
    font-weight: 500;
  }
  /* .communities-item a {
    margin-bottom: 0;
    border-bottom: thin solid ${({ theme }) => theme.lineColor.PostBorder};
    padding-bottom: 16px;
    border-radius: 4px 4px 0 0;
    padding-top: 16px;
    display: flex;
    flex-direction: column;
    margin-top: -1px;
    padding: 16px;
    color: inherit;
    text-decoration: inherit;
  }
  .item {
    align-items: center;

    display: flex;

    justify-content: space-between;
  }
  .item img {
    background-color: #205493;
    font-size: 36px;
    height: 36px;
    max-height: 36px;
    max-width: 36px;
    overflow: hidden;
    width: 36px;
    line-height: normal;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    border-radius: 100%;
    vertical-align: middle;
  }
  .info {
    flex-grow: 1;
    padding: 0 8px;
    overflow: hidden;
    overflow-wrap: break-word;
  }
  .info2 {
    display: flex;
    align-items: baseline;
    flex-direction: column;
  }
  .info2 h6 {
    font-size: 12px;
    font-weight: 500;
    line-height: 16px;
    color: ${({ theme }) => theme.color.titleText_of_link};
    font-weight: 700;
    max-width: 100%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
  .info2 h6:hover {
    text-align: center;
  }
  .info2 p {
    font-size: 12px;
    font-weight: 400;
    line-height: 16px;
    color: #ccc;
  }
  .button {
    flex-shrink: 0;
    width: 88px;
  }
  button {
    width: 100%;
    position: relative;
    background-color: ${({ theme }) => theme.background.active};
    border: 1px solid transparent;
    color: ${({ theme }) => theme.color.secondary};

    font-family: Noto Sans, Arial, sans-serif;
    font-size: 14px;
    font-weight: 700;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
    min-height: 32px;
    min-width: 32px;
    padding: 4px 16px;
    align-items: center;
    border-radius: 9999px;
    box-sizing: border-box;
    display: flex;
    justify-content: center;
    text-align: center;
  }
  button:hover {
    background-color: ${({ theme }) => theme.background.card_hover};
  }

  a {
    text-decoration: none;
  }
  p {
    margin: 0;
    padding: 0;
  }
  h6,
  a,
  h4,
  p {
    margin: 0;
    padding: 0;
  } */
`;
