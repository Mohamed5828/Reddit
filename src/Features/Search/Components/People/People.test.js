import { shallow } from "enzyme";
import People from "./People";
describe("People", () => {
  it("this is a test for People", () => {
    expect(shallow(<People />)).toMatchSnapshot();
  });
});
