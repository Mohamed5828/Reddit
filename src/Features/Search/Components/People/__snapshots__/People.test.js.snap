// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People this is a test for People 1`] = `
<styled.div>
  <p
    className="title"
  >
    People
  </p>
  <PeopleCardItem
    about="A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)"
    avatar="download.jpg"
    key="t5_imagepro"
    totalKarmas={10}
    userID="t5_imagepro"
    username="/u/Abdo"
  />
  <PeopleCardItem
    about="A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)"
    avatar="download.jpg"
    key="t5_imagepro1"
    totalKarmas={3}
    userID="t5_imagepro1"
    username="/u/Abdelrahman"
  />
  <PeopleCardItem
    about="A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)"
    avatar="download.jpg"
    key="t5_imagepro2"
    totalKarmas={1}
    userID="t5_imagepro2"
    username="/u/Ashraf"
  />
  <PeopleCardItem
    about="A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)"
    avatar="download.jpg"
    key="t5_imagepro3"
    totalKarmas={21}
    userID="t5_imagepro3"
    username="/u/Mohamed"
  />
  <PeopleCardItem
    about="A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)"
    avatar="download.jpg"
    key="t5_imagepro4"
    totalKarmas={2100}
    userID="t5_imagepro4"
    username="/u/Elsayed"
  />
</styled.div>
`;
